python/src/aiconfig/editor/client/src/components/AIConfigEditor.tsx

```diff
@@ -633,13 +637,26 @@ export default function EditorContainer({
                 output: event.data,
               });
             } else if (event.type === "aiconfig") {
+              // Next PR: Change this to aiconfig_stream to make it more obvious
+              // and make STREAM_AICONFIG it's own event so we don't need to pass
+              // the `isRunning` state to set. See Ryan's comments about this in
               dispatch({
                 type: "CONSOLIDATE_AICONFIG",
                 action: {
                   ...action,
+                  // Keep the prompt running state until the end of streaming
+                  isRunning: true,
                 },
                 config: event.data,
               });
+            } else if (event.type === "stop_streaming") {
+              // Pass this event at the end of streaming to signal
+              // that the prompt is done running and we're ready
+              // to consolidate the entire AIConfig result
```

Comment:
> and we're ready to consolidate the entire AIConfig result

nit, consolidation can't really happen on this action since there's no aiconfig data to use to consolidate with, right? We could technically return the aiconfig as part of the stop_streaming response and then do consolidation with it, then this comment would be accurate -- and I guess that could be done in place of the final aiconfig response instead of sending an extra response

Reply:
You are correct, let me adjust this comment to say "ready to reset the AIConfig to a non-running state"